pkg/runner/cluster_k8s.go

```diff
@@ -216,7 +246,43 @@ func (c *ClusterK8sRunner) Run(ctx context.Context, input *api.RunInput, ow *rpc
 	var eg errgroup.Group
 
 	eg.Go(func() error {
-		return c.monitorTestplanRunState(ctx, ow, input)
+		var err error
+		err = c.monitorTestplanRunState(ctx, ow, input, &journal)
+
+		lastId := "0"
+
+		for {
+			notifications, newId, errn := notification.Fetch(&template, lastId)
+			if errn != nil {
+				return errn
+			}
```

Comment:
This is a major change. While previously we just returned from `monitorTestplanRunState` which waits for pods to complete (either in `Success` state or in `Failed`, doesn't matter), we now just wait for the test plan processes to exit, and then determine the outcome based on the `notifications`: currently we expect all test plan instances to report `{"outcome": "ok"}` event, and if they don't, we have a failure.